### PR TITLE
Feat(hardware): Difference between hardware listing and hardware details tests counts

### PR DESCRIPTION
**Pull request by Francisco2002**
- [x] Add validation of repeated tests in hardware listing and hardware details
- [x] Treat NULL tests status

**How to test**
-   Compare the counter [hardware google,juniper-sku16](https://staging.dashboard.kernelci.org:9000/hardware?hardwareSearch=google%2Cjuniper-sku) in hardware listing with the counter in hardware details.
-   Test the same hardware counter in [localhost](http://localhost:5173/hardware?hardwareSearch=google%2Cjuniper-sku)

Close of #687

### CODE REVIEW

File: backend/kernelCI_app/views/hardwareDetailsView.py [hunk 1]
```diff
@@ -373,9 +374,11 @@ def sanitize_records(self, records, trees: List, is_all_selected: bool):
                 processed_builds.add(build_id)
                 continue
 
-            should_process_test = self.test_in_filter(test_filter_key, record)
+            should_process_test = not record["id"] in processed_tests \
+                and self.test_in_filter(test_filter_key, record)
```
Comment (MarceloRobert) [inline backend/kernelCI_app/views/hardwareDetailsView.py:378]: Equivalent code with better readability (and follows black formatter):
```suggestion
            should_process_test = (
                self.test_in_filter(test_filter_key, record)
                and record["id"] not in processed_tests
            )
```

**Review by MarceloRobert — APPROVED** [review]
Seems to be working well, just a nit comment

File: backend/kernelCI_app/views/hardwareDetailsView.py (hunk 1 above)
Comment (murilx) [reply backend/kernelCI_app/views/hardwareDetailsView.py:378]: It's good to remember that using `not it` together as suggested by Marcelo (in the cases where it makes sense) is generally better for readability

**Review by murilx — APPROVED** [review]
LGTM

**Pull request merged**